federated_query/optimizer/physical_planner.py

```
@@ -287,6 +312,96 @@ def _plan_join(self, join: Join) -> PhysicalPlanNode:
+    def _and_has_literal_equality(self, predicate: BinaryOp) -> bool:
+        """Recurse into both sides of an AND for a column = literal conjunct."""
+        if self._filter_has_literal_equality(predicate.left):
+            return True
+        return self._filter_has_literal_equality(predicate.right)
+
+    def _is_column_literal_eq(self, predicate: BinaryOp) -> bool:
+        """Whether an equality compares a column reference to a literal."""
+        return {type(predicate.left), type(predicate.right)} == {ColumnRef, Literal}
+
+        return PhysicalNestedLoopJoin(
+            left=left_plan,
```

Ruff (0.15.15)

[error] Undefined name `left_plan`

(F821)